coordinator/src/main/scala/com/socrata/datacoordinator/service/Service.scala

```diff
@@ -221,7 +224,32 @@ class Service(serviceConfig: ServiceConfig,
       "dataset" -> JString(formatDatasetId(name)),
       "schema" -> jsonifySchema(schema))
 
+  // Repeatedly tries to get a new thread by bumping numThreads, using
+  // compareAndSet to make sure our value is valid
+  @annotation.tailrec
+  private def tryGetMutationThread(numTries: Int = 100): Boolean = {
```

Comment:
So if I'm reading this right we end up sleeping a maximum of 10 seconds.  That seems like a pretty small amount of time given how relatively heavyweight mutations are by nature and how failing too quickly can cause unnecessary pain on retries.  

It seems like something on the order of a few minutes might be more reasonable?  Especially if combined with setting a max jetty thread pool as a failsafe, it doesn't look like we have done that yet for dc unless I missed it.